src/resolvers/Mutation/createEvent.ts

```diff
@@ -119,39 +119,35 @@ export const createEvent: MutationResolvers["createEvent"] = async (
     );
   }
 
-  // Creates new event.
-  const createdEvent = await Event.create({
-    ...args.data,
-    creator: currentUser._id,
-    admins: [currentUser._id],
-    organization: organization._id,
-  });
+  let createdEvent;
 
-  if (createdEvent !== null) {
-    await cacheEvents([createdEvent]);
-  }
+  if (
+    !args.data?.recurring ||
+    (args.data?.recurring && args.data?.recurrance == "ONCE")
+  ) {
+    createdEvent = await Event.create({
+      ...args.data,
+      creator: currentUser._id,
+      admins: [currentUser._id],
+      organization: organization._id,
+    });
+
+    if (createdEvent !== null) {
+      await cacheEvents([createdEvent]);
+    }
 
-  await EventAttendee.create({
-    userId: currentUser._id.toString(),
-    eventId: createdEvent._id,
-  });
+    await associateEventWithUser(currentUser, createdEvent);
 
-  /*
-  Adds createdEvent._id to eventAdmin, createdEvents and registeredEvents lists
-  on currentUser's document.
-  */
-  await User.updateOne(
-    {
-      _id: currentUser._id,
-    },
-    {
-      $push: {
-        eventAdmin: createdEvent._id,
-        createdEvents: createdEvent._id,
-        registeredEvents: createdEvent._id,
-      },
-    }
-  );
+    return createdEvent.toObject();
+  }
+
+  if (args.data?.recurring && args.data?.recurrance == "WEEKLY") {
```

Comment:
consider using a switch statement or a strategy pattern for better extensibility. If-else will work but code quality can be improved